### PR TITLE
refactor: #137 new v5 Avatar

### PR DESCRIPTION
**Pull request by nurm717123**
## Context
In order to create a new Avatar for the v5 release, this PR, as part of #137, focuses on developing new Avatar (circle and square variants).

The Avatar component will accept children, allowing consumers to pass either text or an icon. It also accepts shape, size, and color props. For HTML users, it should work with data attributes in combination with the previously mentioned props.

## Pull request checklist
- [x] Avatar (circle/square)

**Detail as per issue below (required):**

fixes: 
- #137 

https://github.com/user-attachments/assets/3ed0d025-b46f-44ff-85b1-b9521e5cc4af

### CODE REVIEW

**Review by kurtdoherty — APPROVED** [review]
Nice and simple! 👏  No blockers, just a few suggestions.

File: src/components/avatar/avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { FC, HTMLAttributes } from 'react'
+import { ElAvatar } from './styles'
+
+export interface AvatarProps extends HTMLAttributes<HTMLSpanElement> {
+  shape?: 'circle' | 'square'
+  size?: 'medium' | 'small'
+  colour?: 'default' | 'purple'
+}
+
+export const Avatar: FC<AvatarProps> = ({ shape, size, colour, children, ...props }) => {
```
[the remark this answers is not in the document]
Comment (nurm717123) [reply src/components/avatar/avatar.tsx:10]: Ah yes, I forgot to put it there, thanks.

File: src/components/avatar/styles.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { styled } from '@linaria/react'
+
+export const ElAvatar = styled.span`
```
[the remark this answers is not in the document]
Comment (nurm717123) [reply src/components/avatar/styles.ts:3]: yes, that is make sense 👍

File: src/components/avatar/styles.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  letter-spacing: var(--letter-spacing-base, -0.15px);
+
+  &[data-shape='square'] {
+    border-radius: var(--corner-lg, 8px);
+  }
+
+  &[data-colour='purple'] {
+    background: var(--fill-action-lightest, #ecf3ff);
+    color: var(--text-action, #4e56ea);
+    svg {
```
[the remark this answers is not in the document]
Comment (nurm717123) [reply src/components/avatar/styles.ts:32]: ok 👌

**Pull request merged**